### DISCUSSION TITLE
Scala 3 version

### DISCUSSION THREAD

**Discussion by jdsalchow** [scala-tsi/scala-tsi#233] [Ideas]
Is there a scala 3 version planned?

**Comment by dhoepelman** [comment]
No plans currently, the macro system was redesigned so it's a bit of work. Will gladly accept contributions though

**Reply by jdsalchow** [reply]
I took a stab at this, see https://github.com/jdsalchow/scala-tsi/tree/scala-3. It works for the features I use. To turn this into a PR there are a few open points:

- Re-enable enums, I didn't feel like solving the scala-reflect usage for getting all values
- Figure out if there's a better way to deal with macro invocation across scala 2 / 3 than to duplicate TSType.scala. If only scala 2 had inline...
- Reimplement cycle detection
- Maybe add some tests

My particular scala 3 migration issue is solved in the current state. If you feel like pushing this upstream let me know how I can help.

**Reply by dhoepelman** [reply]
Thanks a lot @jdsalchow! I took a quick glance and initial impression is it looks good. I'll take a look and see if I can merge it. Feel free to open a PR for the required version updates (which should just be mergeable as-is) and one for the Scala 3 support.

Also thinking to drop 2.12 support with this, it'll make things a lot easier for the enums and literal types.

**Comment by dhoepelman** [comment]
Expanded upon the contribution in https://github.com/scala-tsi/scala-tsi/pull/272. Planning to release as 0.8.0
